packages/@cdktf/hcl2cdk/lib/generation.ts

```diff
@@ -385,6 +385,34 @@ export async function resource(
     mappedConfig.forEach = t.identifier(forEachIteratorName);
   }
 
+  let countIteratorName: string | undefined;
+  if (count) {
+    countIteratorName = variableName(scope, resource, `${key}_count_iterator`);
```

Comment:
Is there a reason this is called count "iterator"? The underlying class is also just named `TerraformCount` and does not share anything with the TerraformIterators. Could we just get rid of `_iterator`?

Reply:
Good point, I'll rename it